document-store/src/main/java/org/hypertrace/core/documentstore/model/options/UpdateOptions.java

```diff
@@ -9,7 +9,8 @@
 @Builder
 public class UpdateOptions {
   public static UpdateOptions DEFAULT_UPDATE_OPTIONS =
-      UpdateOptions.builder().returnDocumentType(AFTER_UPDATE).build();
+      UpdateOptions.builder().returnDocumentType(AFTER_UPDATE).upsert(false).build();
 
   ReturnDocumentType returnDocumentType;
+  boolean upsert;
```

Comment:
In general, I'd try to avoid booleans in API design, especially because they are not extendible. If I were to introduce a 3rd state in the future, I might introduce another boolean flag and then make a combination invalid. 
Rather, we could prefer enums so that introducing a 3rd state becomes straightforward if required.

```suggestion
  MissingDocumentStrategy missingDocumentStrategy;
```

```java
enum MissingDocumentStrategy {
     CREATE_USING_UPDATES,
     SKIP_UPDATES,
}
```
The bonus advantage we get by using enums is the verbosity of the implementation (which can set the base to the different implementors as well as act as a clear mechanism for the callers on what to expect).